docs/sidechain/spec.md

```diff
@@ -0,0 +1,945 @@
+   model is the union of the signer's list. Keeping the signer's list on the
+   account makes this explicit.
+
+4) It's a more natural way to modify sidechain parameters.
+
+## Setting up a sidechain
+
+Setting up a sidechain requires the following:
+
+1) Create a new account for the door account (or use the root account - this can
+   be useful for sidechains). Note, that while more than one sidechain on a door account
+   is supported, is it discouraged.
+
+2) If necessary, setup the trust lines needed for IOUs.
+
+3) Create the sidechain ledger object on each door account.
+
+4) Enable multi-signatures on the two door accounts. These keys much match the
+   keys used by the witness servers. Note that the two door accounts may have
+   different multi-signature lists.
+
+5) Disable the master key, so only the keys on the multi-signature list can
+   control the account.
+
+## Distributing Signature Rewards
+
+When funds are claimed on the destination chain, signatures will be collected on
+the source chain so the signature rewards can be distributed. These rewards will
+be distributed equally between the "reward accounts" for the attestations that
+provided the quorum of signatures on the destination chain that unlocked the
+claim, and the "reward accounts" for the attestations that provided the quorum
+of signatures on the source chain that unlocked the reward pool. If the reward
+amount is not evenly dividable among st the signers, the Mainer is kept by the
+door account.
+
+## Preventing Transaction Replay
+
+Normally, sequence number prevent transaction replay in the XRP ledger. However,
+this sidechain design allows moving funds from an account from transactions not
+sent by that account. All the information to replay these transactions are
+publicly available. This section describes how the different transaction
+prevent certain attacks - including transaction replay attacks.
+
+To successfully run a `SidechainXChainClaim` transaction, the account sending
+the transaction must own the `CrossChainSeqNum` ledger object referenced in the
+witness server's attestation. Since this sequence number is destroyed when the
+funds are successfully moved, the transaction cannot be replayed.
+
+To successfully add witnesses to a `XChainTransferRewardState` and claim the
+reward pool, the reward state ledger object for that transaction must already
+exist. Since that ledger object is destroyed when the reward pool is claimed,
+the transaction cannot be replayed.
+
+To successfully create an account with the `SidechainXChainAccountCreate`
+transaction, the account to be created must not already exist. If the account
+were deletable, this opens up an attack where the account could be created
+multiple times. To prevent this, accounts created with
+`SidechainXChainAccountCreate` are not deletable. Since the account is not
+deletable, the transaction cannot be replayed.
+
+Since the `SidechainXChainTransfer` can contain an optional destination account
```

Comment:
What's the advantage to having the optional destination account? Is it just to save needing that one `XChainClaim` transaction?

Reply:
Yes, that's it. It's convenience. But it also allows us to potentially use x-addresses to send initiating transactions, which is nice (in the future).